gui/src/2D/controls/structuredTextBlock.ts

```diff
@@ -0,0 +1,1295 @@
+};
+
+/**
+ * Class used to create structured text block control
+ */
+export class StructuredTextBlock extends Control {
+    private _structuredText: StructuredText = [];
+    private _textWrapping = TextWrapping.Clip;
+    private _textHorizontalAlignment = Control.HORIZONTAL_ALIGNMENT_CENTER;
+    private _textVerticalAlignment = Control.VERTICAL_ALIGNMENT_CENTER;
+    //private _lines: any[];
+    private _lines: StructuredTextLines;
+    private _resizeToFit: boolean = false;
+
+    private _lineSpacing: ValueAndUnit = new ValueAndUnit(0);
+    private _outlineWidth: number = 0;
+    private _outlineColor: string = "white";
+    private _underline: boolean = false;
+    private _lineThrough: boolean = false;
+    private _frameColor: string = "#a0a0a0";
+    private _frameOutlineWidth: number = 2;
+    private _frameOutlineColor: string = "#606060";
+    private _frameCornerRadius: number = 0;
+
+    private _underlineRelativeY: number = 0.15;
+    private _lineThroughRelativeY: number = -0.3;
+    private _decorationRelativeThickness: number = 0.08;
+
+    private _xOffset: number = 0;
+    private _yOffset: number = 0;
+    private _scrollX: number = 0;
+    private _scrollY: number = 0;
+    private _scrollable: boolean = false;
+
+    // Width and height of the actual content
+    private _contentWidth: number = 0;
+    private _contentHeight: number = 0;
+
+    // Useful for various optimization (e.g. avoiding parsing lines when it shouldn't)
+    private _characterCount: number = 0;
+    private _characterLimit: number = Infinity;
+    private _lastMeasuredWidth: number = 0;
+    private _linesAreDirty: boolean = true;
+
+    private _hasHoverStyle: boolean = false;
+    private _hoveringPart: null | IStructuredTextPart = null;
+    private _hasHref: boolean = false;
+
+    /**
+     * An event triggered after the text is changed
+     */
+    public onTextChangedObservable = new Observable<StructuredTextBlock>();
+
+    /**
+     * An event triggered after the text was broken up into lines and part have metrics computed
+     */
+    public onLinesReadyObservable = new Observable<StructuredTextBlock>();
+
+    /**
+     * An event triggered when there is a part having an href clicked
+     */
+    public onClickHrefObservable = new Observable<HrefObservableData>();
+
+    /**
+     * An event triggered when the mouse enter a part having an href
+     */
+    public onEnterHrefObservable = new Observable<HrefObservableData>();
+
+    /**
+     * An event triggered when the mouse move out of a part having an href
+     */
+    public onOutHrefObservable = new Observable<HrefObservableData>();
+
+    /**
+     * Function used to split a string into words. By default, a string is split at each space character found
+     */
+    public wordSplittingFunction: Nullable<(line: string) => string[]>;
+
+    /**
+    * Gets or sets structured text to display
+    */
+    @serialize()
+    public get structuredText(): StructuredText {
```

Comment:
Why not calling it text? And thus it will be compatible with TextBlock

Reply:
This is something I really want to achieve. We cannot have two "class" of text not being in the same inheritance branch